src/plugins/mer/mergeneraloptionswidget.cpp

```diff
@@ -86,6 +90,45 @@ MerGeneralOptionsWidget::MerGeneralOptionsWidget(QWidget *parent)
     m_ui->askBeforeStartingVmCheckBox->setToolTip(m_ui->askBeforeStartingVmCheckBox->toolTip().arg(Sfdk::Sdk::osVariant()));
     m_ui->askBeforeClosingVmCheckBox->setToolTip(m_ui->askBeforeClosingVmCheckBox->toolTip().arg(Sfdk::Sdk::osVariant()));
     m_ui->rpmValidationInfoLabel->setText(m_ui->rpmValidationInfoLabel->text().arg(Sfdk::Sdk::osVariant()));
+
+    auto setSignElementsEnabled = [](Ui::MerGeneralOptionsWidget *ui, bool enabled) {
+        ui->defaultSigningUserComboBox->setEnabled(enabled);
+        ui->defaultSigningPassphraseFile->setEnabled(enabled);
+    };
+
+    connect(m_ui->signPackagesByDefaultCheckBox, &QCheckBox::toggled, this, [&](bool checked) {
+        setSignElementsEnabled(m_ui, checked);
+    });
+    m_ui->signPackagesByDefaultCheckBox->setChecked(MerSettings::signPackagesByDefault());
+
+    const bool isGpgAvailable = Sfdk::isGpgAvailable();
+    if (!isGpgAvailable) {
+        m_ui->signErrorLabel->setText(tr("GPG utility not found!"));
+        m_ui->signErrorLabel->setStyleSheet(QLatin1String("color: \"yellow\""));
+        m_ui->signErrorLabel->setVisible(true);
+
+        m_ui->signPackagesByDefaultCheckBox->setChecked(false);
+        m_ui->signPackagesByDefaultCheckBox->setEnabled(false);
+    }
+
+    QList<GpgKeyInfo> availableKeys;
+    bool ok;
+    QString error;
+    execAsynchronous(std::tie(ok, availableKeys, error), Sfdk::availableGpgKeys);
+    if (!ok)
+        Core::MessageManager::write(error, Core::MessageManager::Flash);
+
+    auto signingUsersModel = new SigningUserListModel(availableKeys);
+    m_ui->defaultSigningUserComboBox->setModel(signingUsersModel);
+    const int index = signingUsersModel->getIndexBy(MerSettings::signingUser());
+    m_ui->defaultSigningUserComboBox->setCurrentIndex(index);
```

Comment:
With `GpgKeyInfo::toString/fromString` we could easily fill the combo box with strings and then get current with `GpgKeyIngo::fromString(combo->currentText())`. You know, the most reliable- and easiest to maintain- code is that one we never wrote, so if we can go without model... :)